src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -181,7 +182,23 @@ public function normalize($object, $format = null, array $context = [])
                 continue;
             }
 
-            $attributeValue = $this->getAttributeValue($object, $attribute, $format, $context);
+            try {
+                $attributeValue = $this->getAttributeValue($object, $attribute, $format, $context);
+            } catch (AccessException $e) {
+                if (sprintf('The property "%s::$%s" is not initialized.', \get_class($object), $attribute) === $e->getMessage()) {
```

Comment:
rather than detecting the error message (which could easily be broken if we reword the message, especially as there is no tests covering this), shouldn't we use a subclass of the AccessException instead ?
The rule until now is that the exception _messages_ are not covered by any BC rules and should not be relied upon to catch exceptions.
And this is a cross-component dependency, which would require actually changing the BC rules to allow the current pattern

Reply:
Yes I agree, that's what we've done on 5.3 already, but introducing a new class in 4.4 doesn't look like a good idea to me...